libhb/hb.c

```diff
@@ -1119,6 +1120,12 @@ hb_image_t * hb_get_preview3(hb_handle_t * h, int picture,
  */
 int hb_detect_comb( hb_buffer_t * buf, int color_equal, int color_diff, int threshold, int prog_equal, int prog_diff, int prog_threshold )
 {
+#if HB_PROJECT_FEATURE_NVENC
```

Comment:
We should really need this.

If there is a filter in the chain, we should allow the memory copies that are needed to occur to allow the filter chain to operate. 

Typically the scenarios I would expect are:

1.  Decode (Hardware) -> Encode (Hardware or Software)
2. Decode (Hardware) -> Filters  (Software) -> Encode (Hardware or Software)

In the latter case, we don't get zero copy which is fine. While the filters will likely heavily bottleneck the NVEnc encoder,  not putting decode on the CPU frees up that time to go to the filter so it should still be a benefit and something I'd expect users to use.